bokehjs/src/lib/models/glyphs/webgl/marker.frag

```diff
@@ -328,16 +329,53 @@ float marker_distance(in vec2 p, in int line_cap, in int line_join)
 }
 #endif
 
-#if defined(USE_SQUARE) || defined(USE_SQUARE_CROSS) || defined(USE_SQUARE_DOT) || \
-    defined(USE_SQUARE_X)
+#if defined(USE_RECT)
+// From https://www.shadertoy.com/view/4llXD7 (MIT licensed)
+float rounded_box(in vec2 p, in vec2 size, in vec4 r)
+{
+  vec2 half_size = size/2.0;
+
+  // tl r.x x=-1 y=-1
+  // tr r.y x=+1 y=-1
+  // br r.z x=+1 y=+1
+  // bl r.w x=-1 y=+1
+  vec2 rh = p.x > 0.0 ? r.yz : r.xw;
+  float rq = p.y > 0.0 ? rh.y : rh.x;
+
+  // special case for corner miter joins
+  if (rq == 0.0) {
+    vec2 q = abs(p) - half_size;
+    return max(q.x, q.y);
+  } else {
+    vec2 q = abs(p) - half_size + rq;
+    return min(max(q.x, q.y), 0.0) + length(max(q, 0.0)) - rq;
+  }
+}
+
+float marker_distance(in vec2 p, in int line_cap, in int line_join)
+{
+  float dist = rounded_box(p, v_size, v_border_radius);
+
+  if (line_join != miter_join) {
+    vec2 p2 = abs(p) - v_size/2.0; // Offset from corner
+    dist = max(dist, line_join_distance_no_miter(
+      p2, vec2(0.0, 0.0), vec2(1.0/SQRT2, 1.0/SQRT2), v_linewidth/(2.0*SQRT2), line_join));
+  }
+
+  return dist;
```

Comment:
@ianthomas23, can you review changes to webgl?

Reply:
WebGL shader code looks good to me. All such code looks obfuscated at first glance as it is not immediately obvious what it is doing, so it is just a question of how compact to make it and how much to document it.